models/articles.models.js

```diff
@@ -1,20 +1,51 @@
 const db = require("../db/connection");
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+exports.fetchArticles = (
+  sort_by = "created_at",
+  order = "DESC",
+  topic = undefined
+) => {
+  if (topic !== undefined && order !== undefined && sort_by !== undefined) {
+    return db
+      .query(
+        `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+   FROM articles LEFT JOIN comments
+   ON articles.article_id = comments.article_id
+   WHERE topic = $1
+   GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`,
+        [topic]
+      )
+      .then(({ rows }) => {
+        if (rows.length === 0) {
+          return Promise.reject({ status: 404, msg: "404 Not Found" });
+        } else {
+          const deleteUnuseItem = rows.map((each) => {
+            delete each.body;
+            return each;
+          });
+          return deleteUnuseItem;
+        }
+      });
+  } else if (topic === undefined) {
```

Comment:
This method's working, if you wanted to improve the code and avoid repetition you could add the topic to the query if it is specified (i.e. have a query string that has the WHERE topic = $1 appended to it, depending on whether or not it has been defined.)🙂